packages/mostly-equal/src/test/mostly-equal-matchers.unit.ts

```diff
@@ -0,0 +1,169 @@
+      expect(() => {
+        expect({}).to.mostlyEqual({
+          a: defined,
+        });
+      }).to.throw('expected undefined not to be undefined');
+    });
+  });
+
+  describe('defineUnique', () => {
+    it('should throw if value is not unique', () => {
+      const id = defineUnique('id');
+      expect(() => {
+        expect({
+          a: 'a',
+          b: 'b',
+          c: 'a',
+        }).to.mostlyEqual({
+          a: id,
+          b: id,
+          c: id,
+        });
+      }).to.throw('id - is not unique');
+    });
+    it('should not throw if value is unique', () => {
+      const id = defineUnique('id');
+      expect(() => {
+        expect({
+          a: 'a',
+          b: 'b',
+          c: 'c',
+        }).to.mostlyEqual({
+          a: id,
+          b: id,
+          c: id,
+        });
+      }).to.not.throw();
+    });
+    it('should allow undefined if allowUndefined is set to true', () => {
+      const id = defineUnique('id', true);
+      expect(() => {
+        expect({
+          a: 'a',
+          b: 'b',
+          c: undefined,
+        }).to.mostlyEqual({
+          a: id,
+          b: id,
+          c: id,
+          d: id,
+        });
+      }).to.not.throw();
+    });
+  });
+  describe('defineSame', () => {
+    it('should throw if value is not equal', () => {
+      const id = defineSame('ids');
+      expect(() => {
+        expect({
+          a: 'a',
+          b: 'c',
+          c: 'a',
+        }).to.mostlyEqual({
+          a: id,
+          b: id,
+          c: id,
+        });
+      }).to.throw('ids - are not equal');
+    });
+    it('should not throw if value is equal', () => {
+      const id = defineSame('id');
+      expect(() => {
+        expect({
+          a: 'a',
+          b: 'a',
+          c: 'a',
+        }).to.mostlyEqual({
+          a: id,
+          b: id,
+          c: id,
+        });
+      }).to.not.throw();
+    });
+    it('should allow undefined if allowUndefined is set to true', () => {
+      const id = defineSame('ids', true);
+      expect(() => {
+        expect({
+          a: 'a',
+          b: 'a',
+          c: undefined,
+        }).to.mostlyEqual({
+          a: id,
+          b: id,
+          c: id,
+          d: id,
+        });
+      }).to.not.throw();
+    });
+  });
+  describe('equal', () => {
+    it('should throw if not values are not stickly equal', () => {
+      expect(() => {
+        expect({
+          a: {},
+        }).to.mostlyEqual({
+          a: equal({}),
+        });
+      }).to.throw('expected {} to equal {}');
+    });
+    it('should ot throw if not values are strictly equal', () => {
+      const obj = {};
+
+      expect(() => {
+        expect({
+          a: obj,
+        }).to.mostlyEqual({
+          a: equal(obj),
+        });
+      }).not.to.throw();
+    });
+    it('if stricly equal and truncateData is set to true returns a thumbs up instead of content', () => {
```

Comment:
first, i'd switch it around
```
it('returns a thumbs up instead of content, if stricly equal and truncateData is set to true' ...
```
what does "if strictly equal" means? the `equal()` usage? perhaps it needs a rename?
I see no truncateData explictly set in this assertion, so is this the default behavior? if so, no need to specify it in the it.

Reply:
done